model_one/cli.py

```diff
@@ -1,198 +1,274 @@
-    m = _create(data)
-    m.save(filename)
-    return m
-
-def create_generator(filename: str = None, train_iters : int = None):
-    model = {'model_type': 'generator'}
-    if train_iters:
-        model.update({'model_params': {'train_iters': train_iters}})
-    return _create_or_load(model, filename)
-
-def create_classifier(filename: str = None, train_iters : int = None):
-    model = {'model_type': 'classifier'}
-    if train_iters:
-        model.update({'model_params': {'train_iters': train_iters}})
-    return _create_or_load(model, filename)
-
-
-def models():
-    r = requests.get(_models_api(), headers=_get_auth_header())
-    if r.status_code != 200:
-        raise Exception(r.text)
-    res = r.json()
-    res = []
-    if 'models' in res:
-        for k in res['models']:
-            res.append(BeyondmlModel(k))
-    return res
-
-
-def get_model(id):
-    r = requests.get('{}/{}/status'.format(_models_api(), id),
-                     headers=_get_auth_header())
-    if r.status_code != 200:
-        raise Exception(r.text)
-    return BeyondmlModel(r.json())
-
-
-def train_generator(filename: str, train_X, train_y, validate_X, validate_y, train_iters : int = None):
-    model = create_generator(filename, train_iters)
-    return model._upload_fit(train_X, train_y, validate_X, validate_y)
-
-
-def train_classifier(filename: str, train_X, train_y, validate_X, validate_y, train_iters : int = None):
-    model = create_classifier(filename, train_iters)
-    return model._upload_fit(train_X, train_y, validate_X, validate_y)
+        
+        if self.status is not BeyondmlModelStatus.DATASETS_LOADED:
+            raise BeyondmlModelException("Dataset is required")
+
+        return BeyondmlAPI.fit(self._id)
+
+    @inited
+    def generate(self, input: str):
+        r = BeyondmlAPI.generate(self._id, input)
+        return r["answer"]["responses"][0]["response"]
+
+    @inited
+    def classify(self, input: str):
+        r = BeyondmlAPI.classify(self._id, input)
+        return r["answer"]["scores"]
+
+    @inited
+    def upload(
+        self,
+        train_X: Union[list, Series], 
+        train_y: Union[list, Series],
+        validate_X: Union[list, Series],
+        validate_y: Union[list, Series]
+    ) -> dict:
+        if any(len(data) < MINIMUM_ENTRIES for data in [
+            train_X, train_y, validate_X, validate_y
+        ]):
+            raise BeyondmlModelException(f"Dataset must contain at least {MINIMUM_ENTRIES} elements")
+        
+        data = {
+            "train_dataset": {
+                "inputs": train_X,
+            },
+            "validate_dataset": {
+                "inputs": validate_X,
+            }
+        }
+
+        key = {
+            BeyondmlModelType.GENERATOR: "outputs",
+            BeyondmlModelType.CLASSIFIER: "classes",
+        }[self.type]
+
+        data["train_dataset"][key] = train_y
+        data["validate_dataset"][key] = validate_y
+
+        def _default(val):
```

Comment:
This function just converts pandas.Series to lists during serialization
If you decide to allow usage of numpy.ndarray, conversion to list would be here